client/my-sites/controller.js

```diff
@@ -142,26 +141,26 @@ function isPathAllowedForDomainOnlySite( pathname, domainName ) {
 
 function onSelectedSiteAvailable( context ) {
 	const selectedSite = sites.getSelectedSite();
-	const state = context.store.getState();
+	const getState = () => context.store.getState();
```

Comment:
Yes, it looks tricky, because of those `dispatch` calls in the middle of the method.